core/opendaq/modulemanager/include/opendaq/module.h

```diff
@@ -114,13 +114,13 @@ DECLARE_OPENDAQ_INTERFACE(IModule, IBaseObject)
 
     /*!
      * @brief Creates and returns a server with the specified server type.
-     * To prevent cyclic reference, we should not use the Instance instead of rootDevice.
      * @param serverTypeId The id of the server type to create. ServerType can be retrieved by calling `getAvailableServerTypes()`.
      * @param config Server configuration. In case of a null value, implementation should use default configuration.
      * @param rootDevice Root device.
+     * @param parent The parent component to which the server attaches.
      * @param[out] server The created server.
      */
-    virtual ErrCode INTERFACE_FUNC createServer(IServer** server, IString* serverTypeId, IDevice* rootDevice, IPropertyObject* config = nullptr) = 0;
+    virtual ErrCode INTERFACE_FUNC createServer(IServer** server, IString* serverTypeId, IDevice* rootDevice, IComponent* parent, IPropertyObject* config = nullptr) = 0;
```

Comment:
Technically, do we ever create a server anywhere other than the "rootDevice/Srv" component? If not, we don't need to specify the parent, as we already have the root device.

Reply:
I've removed the parent from the parameter list, as it is always the "Srv" folder of the attached root device. However, the related changes added to the server deserialization factories might look a bit odd now.